File: src/main/Java/boj_step19/boj1010.java
import java.io.*;
import java.math.BigInteger;
import java.util.StringTokenizer;

public class boj1010 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int t = Integer.parseInt(reader.readLine());
        while (t-- > 0) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            int n = Integer.parseInt(tokenizer.nextToken());
            int m = Integer.parseInt(tokenizer.nextToken());
            BigInteger mfac = BigInteger.valueOf(1);
            BigInteger nfac = BigInteger.valueOf(1);
            for (int i = m - n + 1; i <= m; i++) mfac = mfac.multiply(BigInteger.valueOf(i));
            for (int i = 1; i <= n; i++) nfac = nfac.multiply(BigInteger.valueOf(i));
            writer.append(mfac.divide(nfac) + "\n");
        }
        writer.close();
    }
}

File: src/main/boj_step19/boj1010.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.math.BigInteger

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    repeat(n) {
        var (n, m) = r.readLine().split(" ").map { it.toInt() }
        var mfact = BigInteger.valueOf(1)
        var nfact = BigInteger.valueOf(1)
        if (n > m / 2) n = m - n
        for (i in m - n + 1 .. m) mfact *= i.toBigInteger()
        for (i in 1 .. n) nfact *= i.toBigInteger()
        w.appendLine("${mfact / nfact}")
    }
    w.close()
}